app/services/UpdateOrganisationDetailsService.scala

```diff
@@ -51,16 +55,26 @@ class UpdateOrganisationDetailsService @Inject()(updateVatSubscriptionConnector:
     UpdateVatSubscription(
       controlInformation = ControlInformation(welshIndicator),
       requestedChanges = ChangeOrganisationDetailsRequest,
-      organisationDetails = Some(UpdatedOrganisationDetails(None, None, Some(removedNameIsEmpty(updatedTradingName.tradingName)))),
+      organisationDetails = Some(UpdatedOrganisationDetails(None, None, Some(tradingName(updatedTradingName, customerDetails)))),
       updatedPPOB = None,
       updatedReturnPeriod = None,
       updateDeregistrationInfo = None,
       declaration = Declaration(agentOrCapacitor, Signing()),
       commsPreference = None
     )
-  }
+}
 
-  private def removedNameIsEmpty(newTradingName: Option[String]): String = newTradingName.fold("")(name => name)
+  private[services] def tradingName(newTradingName: TradingName, details: CustomerDetails): String = {
+    newTradingName.tradingName.fold {
+      details.organisationName match {
+        case Some(businessName) => businessName
+        case _ =>
+          details.firstName.fold("")(name => name + " ") +
```

Comment:
In the scenario where we only get firstName, or firstName + middleName (unsure of how possible that is but due to optionals we should consider it), there would be a whitespace at the end of the name. Could you use the method `.trim` on the String produced by this logic to remove the trailing whitespace. It wouldn't hurt to add in a unit test for this scenario as well.